File: 58_LengthOfLastWord.py
import re
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        if not s:
            return 0
        split_string = re.split('\s+', s)
        if split_string[-1] == '':
            i = 0
            while split_string[-1-i] == '' and i < len(split_string)-1:
                i += 1
            return len(split_string[-1-i])
        else:
            return len(split_string[-1])

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        if not s:
            return 0
        
        i, s = 0, ' ' + s
        
        while s[-1-i] == ' ' and i < len(s)-1:
            i += 1
            
        if i < len(s)-1:
            j = 0
            while s[-1-i-j] != ' ':
                j += 1
            return j
        else:
            return 0

File: java/58_LengthOfLastWord.java
class Solution {
    public int lengthOfLastWord(String s) {
        int i = s.length(), length = 0;
        
        while (i > 0) {
            i--;
            if (s.charAt(i) != ' ')
                ++length;
            else if (length > 0)
                return length;
        }
        return length;
    }
}
